internal/batches/service/service.go

```diff
@@ -380,9 +381,43 @@ func (svc *Service) ParseBatchSpec(data []byte) (*batcheslib.BatchSpec, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "parsing batch spec")
 	}
+	if err = handleMount(dir, spec); err != nil {
+		return nil, errors.Wrap(err, "parsing batch spec")
+	}
 	return spec, nil
 }
 
+func handleMount(dir string, spec *batcheslib.BatchSpec) error {
+	for i, step := range spec.Steps {
+		for j, mount := range step.Mount {
+			p := mount.Path
+			if !filepath.IsAbs(p) {
+				// Try to build the absolute path since Docker will only mount absolute paths
+				p = filepath.Join(dir, p)
+			}
+			pathInfo, err := os.Stat(p)
+			if os.IsNotExist(err) {
+				return errors.Newf("step %d mount path %s does not exist", i+1, p)
+			} else if err != nil {
+				return errors.Wrapf(err, "step %d mount path validation", i+1)
+			} else {
+				if !strings.HasPrefix(p, dir) {
```

Comment:
I'm wondering if we'll have edge cases here around symlinks — for example, if the user has their spec in `/Users/foo/bar`, but there's a symlink `/Users/foo/quux` to it and they use the latter in an absolute path provided to `mount`.

The quick and easy solution here might be to call [`filepath.EvalSymlinks`](https://pkg.go.dev/path/filepath#EvalSymlinks) on `p` and `dir` before performing this check. (I guess we only really need to do `dir` once.)

Reply:
Hmm, don't you think that could actually be intentional? 

Like I put a symlink down to use it in a different WD without copying it?

Or do I misunderstand what the problem is here?

Reply:
I guess my question is more: should we consider `/Users/foo/bar` and `/Users/foo/quux` to be the same for the purposes of this check, if they point to the same directory?

Reply:
I, having thought about this at least an entire 30 seconds would say we should not. 

If you symlink `/Users/foo/spec.yml` into `/Users/erik/workspace/spec.yml` and use that one, then the intentional workspace location IMO would be `/Users/erik/workspace` and not `/Users/foo`. I only symlinked the spec file and not it's surroundings. If I symlinked the entire directory `/Users/foo` to `/Users/erik/workspace`, then I should just be using relative paths or paths to the aliased name. 
At least to my 12:30am brain that sounds logical. Please challenge me 😬